### PR TITLE
Warnings support

### LINKED ISSUE #16: Support for warnings
Does the library support the `_warning` field that can come back? This is documented here:

https://www.geocod.io/docs/#warnings

As an example:

```shell
curl --request GET \
  --url 'https://api.geocod.io/v1.7/geocode?api_key=<SNIPPED>&fields=acs-social&q=Road%20X%2C%20Hot%20Springs%2C%20AR&limit=1'
```

returns

```json
{
	"input": {
		"address_components": {
			"street": "Rd",
			"formatted_street": "Rd",
			"city": "Hot Springs",
			"state": "AR",
			"country": "US"
		},
		"formatted_address": "Rd, Hot Springs, AR"
	},
	"results": [
		{
			"address_components": {
				"city": "Hot Springs",
				"county": "Garland County",
				"state": "AR",
				"zip": "71901",
				"country": "US"
			},
			"formatted_address": "Hot Springs, AR 71901",
			"location": {
				"lat": 34.526453,
				"lng": -92.974072
			},
			"accuracy": 0.45,
			"accuracy_type": "place",
			"source": "TIGER\/Line® dataset from the US Census Bureau",
			"fields": {
				"census": {
					"2021": {
						"census_year": 2021,
						"state_fips": "05",
						"county_fips": "05051",
						"tract_code": "010502",
						"block_code": "1001",
						"block_group": "1",
						"full_fips": "050510105021001",
						"place": null,
						"metro_micro_statistical_area": {
							"name": "Hot Springs, AR",
							"area_code": "26300",
							"type": "metropolitan"
						},
						"combined_statistical_area": {
							"name": "Hot Springs-Malvern, AR",
							"area_code": "284"
						},
						"metropolitan_division": null,
						"source": "US Census Bureau"
					}
				}
			},
			"_warnings": [
				"acs-social was skipped since result is not street-level"
			]
		}
	]
}
```

I don't see any way to read the `_warnings` field.

In addition, the documentation states the `_warnings` field can also show up on the same level as `results`.

```json
{
  "input": {
    ...
  },
  "results": [
    ...
  ],
  "_warnings": [
    "Ignoring parameter \"postalcode\" as it was not expected. Did you mean \"postal_code\"? See full list of valid parameters here: https://www.geocod.io/docs/"
  ]
}
```

I assume there just needs to be new fields added to `GeocodingResponse` and `GeocodingResult`.

### PR DESCRIPTION
**Pull request by ctmay4**
Closes #16 

Add support for the `_warnings` field as defined in https://www.geocod.io/docs/#warnings

### CODE REVIEW

File: src/test/java/io/github/deansg/jeocodio/IntegrationTest.java [hunk 1]
```diff
@@ -19,12 +23,28 @@ public void setUp() {
...
 
         var result = future.get();
         Assertions.assertNotNull(result);
         Assertions.assertEquals(1, result.results().size());
     }
+
+    @Disabled("Should be used for local testing with a real API key as an environment variable")
+    @Test
+    public void testGeocodingE2EWarnings() throws ExecutionException, InterruptedException {
+        // test warnings
```
Comment (deansg) [inline src/test/java/io/github/deansg/jeocodio/IntegrationTest.java:38]: Please remove the comment, the test name is indicative enough

File: src/test/java/io/github/deansg/jeocodio/IntegrationTest.java [hunk 2]
```diff
@@ -19,12 +23,28 @@ public void setUp() {
...
+        var geocodingRequest = GeocodingRequestBuilder.builder()
+                .q("Road X, Hot Springs, AR")
+                .fields(List.of("acs-social"))
+                .limit(1)
+                .build();
+        var result = geocodioClient.geocodeAsync(geocodingRequest).get();
```
Comment (deansg) [inline src/test/java/io/github/deansg/jeocodio/IntegrationTest.java:44]: please add blank lines that separate the 3 stages of the test (arrange/act/assert) as done in the other test in this file

**Review by deansg — COMMENTED** [review]
2 more general comments:
1. I know it's out-of-scope for your change, but if it's easy for you can you add a condition to .github/workflows/test.yml so that the tests job runs on external pull requests as well? You just need to modify the on clause
2. If possible, can you write a very basic unit test for one of the changed models that just checks its serialization and deserialization from a string? To make sure the change you made works

File: src/test/java/io/github/deansg/jeocodio/IntegrationTest.java (hunk 2 above)
Comment (ctmay4) [reply src/test/java/io/github/deansg/jeocodio/IntegrationTest.java:44]: ok

**Comment by ctmay4** [comment]
I updated the workflow but it looks like the action doesn't trigger until after the approval.

**Comment by deansg** [comment]
> I updated the workflow but it looks like the action doesn't trigger until after the approval.

It needed me to approve the addition of a workflow (disables others from running malicious actions merely by opening a pull request).

**Review by deansg — APPROVED** [review]

**Pull request merged**